org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurDialog.java

```diff
@@ -242,10 +258,15 @@ public JDialog createDialog(Component parent) throws HeadlessException {
 				open.setDialogTitle(isOpen ? "Open..." : "Save...");
 				open.setApproveButtonText(isOpen ? "Open" : "Save");
 				open.setDialogType(isOpen ? JFileChooser.OPEN_DIALOG : JFileChooser.SAVE_DIALOG);
-				if (ext.length() > 0)
+				if (exts != null && exts.length > 0)
 					open.setFileFilter(new FileFilter() {
 						public boolean accept(File file) {
-							return !file.isFile() || file.getPath().toLowerCase(Locale.US).endsWith(ext);
+							for ( String ext : exts) {
```

Comment:
Replace this `for` loop with something like (I probably got the syntax wrong): `!file.isFile() || exts.toStream().any(ext -> endsWithIgnoreCase(file.getPath(), ext));`, where `endsWithIgnoreCase` is a helper static method method.

I suggested extracting helper method `endsWithIgnoreCase` because that logic is used at least twice in this method, and the way it is implemented it only converts the LHS of `endsWith` to lower case although it seems that the RHS should be converted to lower case too (e.g., `str.toLowerCase(Locale.US).endsWith(ext.ToLowerCase(Locale.US))`)

If you really insist on keeping the for loop, then `!file.isFile()` can be moved out of the loop.